.github/workflows/code.yml

```diff
@@ -53,7 +64,7 @@ jobs:
           python-version: ${{ matrix.python }}
 
       - name: Install with latest dependencies
-        run: pip install .[dev]
+        run: .github/workflows/pip_install.sh requirements-test-${{ matrix.python }}-${{ matrix.os }}.txt ${{ matrix.install }}
```

Comment:
It may be neater to name these files `requirements-test-${{ matrix.os }}-${{ matrix.python }}.txt`, so that they are sorted by os and then python. That precedence makes more sense to me.